### DISCUSSION TITLE
How can we estimate the size ./pgsql_tmp will have on importing from a OSM postgres db ?

### DISCUSSION THREAD

**Discussion by stouch** [komoot/photon#725] [Q&A] [answered]
I am indexing Photon data from a nominatim postgres db, and I have some storage constraints (postgres db, and photon data, have their own storage volumes). I noticed that on `java -jar photon-*.jar -nominatim-import`, the postgres tmp file becomes quickly huge, so I'd like to estimate how much size it's gonna take.

Thanks,

Regards,

**Comment by stouch** [comment, marked as answer]
Ok so, for this setup (12 west-european + 3 north-african countries, in 8 languages), after a nominatim install without updates (multiple --osm-file) and with a flatnode file configured, I've got : 
``` 
80G   ./nominatim-project/flatnode
275G  ./postgres-data/main
``` 
with a : 
27G	./postgres-data/main/base/pgsql_tmp

so about ~10% of the total postgresql db


Do you think it's what I should expect ?

**Reply by lonvia** [reply]
I don't have exact numbers but it sounds about right. Our export server seems to get around 100GB for the planet (around 800GB postgres size).

Photon sorts the data for import to be able to save on the number of lookups in the Postgres DB. The sorting creates that temporary file.
